src/main/java/net/snowflake/client/core/HttpClientSettingsKey.java

```diff
@@ -15,35 +15,34 @@
 public class HttpClientSettingsKey implements Serializable {
 
   private OCSPMode ocspMode;
-  private boolean useProxy;
+  private boolean useProxy = false;
   private String proxyHost = "";
   private int proxyPort = 0;
   private String nonProxyHosts = "";
   private String proxyUser = "";
   private String proxyPassword = "";
   private String proxyProtocol = "http";
+  private String userAgentSuffix = "";
 
   public HttpClientSettingsKey(
       OCSPMode mode,
+      boolean useProxy,
       String host,
       int port,
       String nonProxyHosts,
       String user,
       String password,
-      String scheme) {
-    this.useProxy = true;
+      String scheme,
+      String userPrefix) {
+    this.useProxy = useProxy;
     this.ocspMode = mode != null ? mode : OCSPMode.FAIL_OPEN;
     this.proxyHost = !Strings.isNullOrEmpty(host) ? host.trim() : "";
     this.proxyPort = port;
     this.nonProxyHosts = !Strings.isNullOrEmpty(nonProxyHosts) ? nonProxyHosts.trim() : "";
     this.proxyUser = !Strings.isNullOrEmpty(user) ? user.trim() : "";
     this.proxyPassword = !Strings.isNullOrEmpty(password) ? password.trim() : "";
     this.proxyProtocol = !Strings.isNullOrEmpty(scheme) ? scheme.trim() : "http";
-  }
-
-  public HttpClientSettingsKey(OCSPMode mode) {
```

Comment:
Please don't get rid of public method. Check with Sashi for Deprecation process, if needed.

Reply:
@sfc-gh-sbommakanty, could you help understand if it's ok to refactor this code and the deprecation process?

Reply:
Kept existing public methods.